# Model Baseline: Heurística

> **Dataset:** Stroke Prediction Dataset (Kaggle)  
> **Fuente:** https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset  
> **Algoritmo principal:** Heurística basada en EDA  
> **Búsqueda de hiperparámetros:** No aplica 

In [17]:
import sys
sys.path.insert(0, "..")

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import GridSearchCV, cross_val_score

from src.data_loader import load_dataset, get_train_test_split
from src.evaluation import evaluate_model, plot_full_evaluation
from src.model_registry import save_model

from sklearn.metrics import classification_report

In [13]:
df = load_dataset()
X_train, X_test, y_train, y_test = get_train_test_split(df)

Loaded dataset: 5110 rows, 12 columns.
Split: train=4088 rows, test=1022 rows (test_size=0.2, seed=13)


In [14]:
# Heurística simple basada en EDA y análisis de características

def baseline_rule(row):
    score = 0
    
    if row['age'] > 60:
        score += 1
    if row['hypertension'] == 1:
        score += 1
    if row['heart_disease'] == 1:
        score += 1
    if row['avg_glucose_level'] > 140:
        score += 1
    
    return 1 if score >= 2 else 0

In [15]:
y_pred_baseline = X_test.apply(baseline_rule, axis=1)

In [18]:
print(classification_report(y_test, y_pred_baseline))

              precision    recall  f1-score   support

           0       0.97      0.89      0.93       972
           1       0.17      0.44      0.24        50

    accuracy                           0.86      1022
   macro avg       0.57      0.66      0.58      1022
weighted avg       0.93      0.86      0.89      1022



El modelo baseline basado en reglas simples logra una alta precisión para la clase mayoritaria (0), con un valor de 0.97, y un desempeño global aceptable en términos de accuracy (0.86). Sin embargo, su capacidad para detectar la clase minoritaria (stroke) es limitada, alcanzando un recall de 0.44 y una precisión de 0.17, lo que evidencia una cantidad considerable de falsos positivos y falsos negativos.

A pesar de su simplicidad, el modelo logra capturar parcialmente patrones relevantes asociados al riesgo de stroke, constituyendo una referencia inicial útil. No obstante, su bajo desempeño en la detección de la clase positiva justifica la necesidad de utilizar modelos más complejos que permitan mejorar la sensibilidad sin deteriorar significativamente la precisión.